akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala

```diff
@@ -150,8 +149,96 @@ class CoordinatedShutdownSpec
         testActor ! "C"
         Future.successful(Done)
       }
-      Await.result(co.run(UnknownReason), remainingOrDefault)
-      receiveN(4) should ===(List("A", "B", "B", "C"))
+      whenReady(co.run(UnknownReason)) { _ =>
+        receiveN(4) should ===(List("A", "B", "B", "C"))
+      }
+    }
+
+    "cancel tasks" in {
+      import system.dispatcher
+      val phases = Map("a" -> emptyPhase, "b" -> phase("a"))
+      val co = new CoordinatedShutdown(extSys, phases)
+
+      val task0Counter = new AtomicInteger()
+
+      val task0: () => Future[Done] = () =>
+        Future {
+          task0Counter.incrementAndGet()
```

Comment:
I think it would be cleaner if we used `TestProbe` or the `testActor`  throughout instead of mixing `AtomicInteger` and `Promise`s to verify that the tasks were run (like the other pre-existing tests do).

Reply:
Can do. My concern was using an actor while also possibly shutting down the actor system it's on, but that doesn't seem to be happening.